tests/test_carrier_account.py

```diff
@@ -118,11 +120,14 @@ def test_carrier_account_create_ups(prod_client):
 
 
 @pytest.mark.vcr()
-def test_carrier_account_update_ups(prod_client):
+@patch("easypost.services.carrier_account_service.CarrierAccountService.retrieve")
```

Comment:
If you look at the cassette, it's using the `/carrier_accounts` endpoint for updates because this is being mocked. We should try to assert 422 down below like what I suggest above.

Reply:
Check again, I forgot to update the return type. Mocking this works as expected. If you look above, I responded as to why using a 422 will not work anymore in a couple weeks and we cannot rely on it.

Reply:
We are creating `DhlEcsAccount` in the test, not UPS, so it should still work in a couple of weeks unless I missed something. Anyway, it’s not a dealbreaker, just something I think we can improve.

Reply:
Yeah I feel like creating a DHL account then attempting to update it view the UPS flow is even more confusing and a worse test because it's mixing things that aren't compatible. I like the thought to get it to 422, I just don't think this particular approach is the right way to do it.